readthedocs/embed/v3/views.py

```diff
@@ -0,0 +1,318 @@
+"""Views for the EmbedAPI v3 app."""
+
+import logging
+import re
+from urllib.parse import urlparse
+import requests
+
+from selectolax.parser import HTMLParser
+from pyquery import PyQuery as PQ  # noqa
+
+from django.conf import settings
+from django.core.cache import cache
+from django.shortcuts import get_object_or_404
+from django.utils.functional import cached_property
+from rest_framework import status
+from rest_framework.permissions import AllowAny
+from rest_framework.renderers import BrowsableAPIRenderer, JSONRenderer
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from readthedocs.api.v2.mixins import CachedResponseMixin
+from readthedocs.core.unresolver import unresolve
+from readthedocs.core.utils.extend import SettingsOverrideObject
+from readthedocs.embed.utils import clean_links
+from readthedocs.projects.constants import PUBLIC
+from readthedocs.storage import build_media_storage
+
+log = logging.getLogger(__name__)
+
+
+
+class EmbedAPIBase(CachedResponseMixin, APIView):
+
+    # pylint: disable=line-too-long
+
+    """
+    Embed a section of content from any Read the Docs page.
+
+    ### Arguments
+
+    * url (with fragment) (required)
+    * doctool
+    * doctoolversion
+
+    ### Example
+
+    GET https://readthedocs.org/api/v3/embed/?url=https://docs.readthedocs.io/en/latest/features.html%23#full-text-search
+
+    """  # noqa
+
+    permission_classes = [AllowAny]
+    renderer_classes = [JSONRenderer, BrowsableAPIRenderer]
+
+    @cached_property
+    def unresolved_url(self):
+        url = self.request.GET.get('url')
+        if not url:
+            return None
+        return unresolve(url)
+
+    def _download_page_content(self, url):
+        cache_key = f'embed-api-{url}'
+        cached_response = cache.get(cache_key)
+        if cached_response:
+            log.debug('Cached response. url=%s', url)
+            return cached_response
+
+        try:
+            response = requests.get(url, timeout=1)
+        except requests.exceptions.TooManyRedirects:
+            log.warning('Too many redirects. url=%s', url)
+            return
+        except Exception:  # noqa
+            log.warning('There was an error reading the URL requested. url=%s', url)
```

Comment:
I'm fine doing `log.exception` here. However, I think this will log a lot in Sentry because people doing weird stuffs. I think we can start logging it first and then removing if we think it's too noisy.